pvlib/tools.py

```diff
@@ -207,7 +207,28 @@ def _pandas_to_doy(pd_object):
     -------
     dayofyear
     """
-    return pd_object.dayofyear
+    return _pandas_to_utc(pd_object).dayofyear
+
+
+def _pandas_to_utc(pd_object):
```

Comment:
No, it is not. I have considered the function but failed to call it instead of `_pandas_to_utc` in `solarposition.py` for three reasons. Firstly, `tools.localize_to_utc` requires `location` as an argument but #2054 should be fixed regardless of `Location`. Another reason is that `tools.localize_to_utc` assumes the ambiguous timezone of a tz-naive datetime as the default local timezone but I wanted to assume it as 'UTC' in the same way as #333. Finally, if I changed the utility function, some of the previous users would get confused, though luckily there seems no internal usage of `tools.localize_to_utc`.